(feat)Cloth code rework

EnderMen3132 opened the pull request:
<!-- Пишите **НИЖЕ** заголовков и **ВЫШЕ** комментариев, иначе ваш текст может не отобразиться. -->
<!-- В Contributing.MD вы можете найти некоторые рекомендации к оформлению пулл-реквеста. -->

## Описание
<!-- Опишите, что делает ваш ПР. Документировать каждую деталь не требуется, просто укажите основные изменения. -->
реворк кода для изменения пути спрайтов и добавления в него строчки кода с временем одевания и раздевания вещи (по умолчанию стоит на 0 тобишь неактивна и не затрагивает обываетелей) 

## Ссылка на предложение/Причина создания ПР
<!-- Здесь оставьте ссылку на сообщение в #отчеты-по-предложениям, чтобы подтвердить, что ваше предложение одобрено. Либо укажите, почему этот ПР должен пройти без предложки. -->
<!-- Пример ссылки: https://discord.com/channels/617003227182792704/755125334097133628/ID-сообщения -->
https://discord.com/channels/617003227182792704/1030133547509485708
## Демонстрация изменений
<!-- Здесь вы можете показать изменения внешне, к примеру новые спрайты, звуки или изменения карты. Или написать, что именно изменилось для игроков. Этот пункт полностью опционален и его можно удалить. -->

EnderMen3132 committed:
Cloth code rework

code rework for future development

EnderMen3132 committed:
Merge branch 'master220' into Cloth-code-rework

EnderMen3132 committed:
исправления

контрибьютор накричал пришлось убирать ненужные коментарии и менять названия для переменной

Furukia reviewed (approved):
Глянула код и указала на проблемы. Разбирали их в дискорде на полигоне.

TrashDoxx commented:
Аа, ээ, а зачем?

Den109G commented:
Быстро пробежался по изменениям.
Пропали спрайты на голове у:
```
/obj/item/trash/fried_vox
/obj/item/nullrod/fedora
/obj/item/storage/bag/plasticbag
/obj/item/reagent_containers/food/snacks/grown/poppy
/obj/item/reagent_containers/food/snacks/grown/moonlight
/obj/item/reagent_containers/food/snacks/grown/harebell
/obj/item/grown/sunflower
Все /obj/item/holder (то есть у мышей, дронов и т.д. на голове пропали иконки)
/obj/item/reagent_containers/glass/bucket
```

Пропали спрайты на спине у:
```
Всех разновидностях нуллрода, что крепятся на спину
/obj/item/melee/energy_katana
/obj/item/toddler
/obj/item/radio/electropack
/obj/item/chrono_eraser
/obj/item/defibrillator
/obj/item/shield/riot
/obj/item/melee/baton/cattleprod
/obj/item/twohanded/fireaxe
Любой баллон
/obj/item/watertank
/obj/item/twohanded/bostaff
/obj/item/twohanded/kinetic_crusher
/obj/item/hierophant_club
/obj/item/lava_staff
/obj/item/shared_storage
/obj/item/gun/energy/lasercannon
/obj/item/gun/energy/pulse
/obj/item/gun/energy/ionrifle
/obj/item/gun/energy/temperature
/obj/item/gun/energy/sniperrifle
Любой /obj/item/gun/magic/staff
/obj/item/gun/projectile/revolver/doublebarrel
/obj/item/gun/projectile/shotgun
/obj/item/gun/projectile/automatic/sniper_rifle
```

И в придачу оно фигово работает с теми предметами, что можно повесить на много слотов (к примеру если починишь энерго катану, то заметишь, что повесив на пояс она будет отображаться на спине, т.к. только один файл использует для отображения)

Den109G commented:
Хорошие намерения, да. Но не понятно зачем.
Да и ещё зачем-то симпл мобам дали изменения, хотя они никак не используются для спрайтов на карбоне.

PhantornRU commented on code/modules/mob/living/simple_animal/friendly/mouse.dm:
```diff
@@ -1,6 +1,7 @@
 /mob/living/simple_animal/mouse
 	name = "mouse"
 	real_name = "mouse"
+	onmob_icon = 'icons/mob/head.dmi'
```
Не пихай это в моба. Пихай это в holder.dm. Это предмет который и является удержанием моба. Не издевайся над мышками.

EnderMen3132 commented:
щас буду делать исправления

EnderMen3132 committed:
Исправления

попытался исправить проблемы на которые мне указали в коментариях, всё ещё не решена проблема с вещами которые можно одеть в 2 слота
возможно остались предметы которым непрописан путь к спрайтам

Den109G commented on code/game/objects/items.dm:
```diff
@@ -3,6 +3,7 @@ GLOBAL_DATUM_INIT(fire_overlay, /image, image("icon" = 'icons/goonstation/effect
 	var/onmob_icon  //onmob_icon отвечает за место поиска спрайта на вашего персонажа пример:"icon/mob/вашфайл.dmi" и его можно вписать отдельно для каждой вещи
 	name = "item"
 	icon = 'icons/obj/items.dmi'
+	onmob_icon = 'icons/mob/back.dmi'
```
А почему не сразу `var/onmob_icon = 'icons/mob/back.dmi'`?

EnderMen3132 replied:
решил так сделать

EnderMen3132 replied:
пытался исправить проблему с тем что предметы на 2 слота не меняют спрайт но чёт задушился
